autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/coalescer/base_coalescer.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Dict, List
+
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.base_tabular_preprocessing import (
+    autoPyTorchTabularPreprocessingComponent
+)
+from autoPyTorch.utils.common import FitRequirement
+
+
+class BaseCoalescer(autoPyTorchTabularPreprocessingComponent):
+    def __init__(self) -> None:
+        super().__init__()
+        self.add_fit_requirements([
+            FitRequirement('categorical_columns', (List,), user_defined=True, dataset_property=True),
+            FitRequirement('categories', (List,), user_defined=True, dataset_property=True)
+        ])
+
+    def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Add the preprocessor to the provided fit dictionary `X`.
+
+        Args:
+            X (Dict[str, Any]): fit dictionary in sklearn
+
+        Returns:
+            X (Dict[str, Any]): the updated fit dictionary
+        """
+        if all(self.preprocessor[key] is None for key in ['numerical', 'categorical']):
```

Comment:
```suggestion
        if all(self.preprocessor[key] is None for key in ['categorical']):
```

Reply:
actually as we are only building a categorical preprocessor `self.preprocessor['numerical'` will always be none